_posts/en/newsletters/2023-10-25-newsletter.md

```diff
@@ -0,0 +1,506 @@
+      so Bob would never actually need to pay higher and higher
+      feerates.  Whether that would be true in practice on the Bitcoin
+      network is unproven.
+
+    - **Automatic retrying of past transactions:** Corallo
+      [suggested][corallo cycle1] that, "the only fix for this issue
+      will be when miners keep a history of transactions they've seen
+      and try them again after [...] an attack like this."  Bastien
+      Teinturier [replied][teinturier cycle], "I agree with Matt though
+      that more fundamental work most likely needs to happen at the
+      bitcoin layer to allow L2 protocols to be more robust against that
+      class of attacks."  Riard also [said][riard cycle3] something
+      similar, "a sustainable fix can [only] happen at the base-layer,
+      e.g adding a memory-intensive history of all seen transactions".
+
+    - **Presigned fee bumps:** Peter Todd [argued][todd cycle1] that,
+      "the correct way to do pre-signed transactions is to pre-sign
+      enough *different* transactions to cover all reasonable needs for
+      bumping fees. [...] There is zero reason why the B->C transactions
+      should be getting stuck."  (Emphasis in the original.)
+
+        That could work something like this: for the HTLC between Bob
+        and MalloryB, Bob gives MalloryB ten different signatures for
+        the same preimage spend at different feerates.  Note that this
+        doesn't require that MalloryB disclose the preimage to Bob at
+        signing time.  At the same time, MalloryB gives Bob ten
+        different signatures for the same refund spend at different
+        feerates.  This can be done before the refund can be broadcast.
+        The feerates used might be (in sats/vbyte): 1, 2, 4, 8, 16, 32,
+        64, 128, 256, 512, 1024, which should cover anything for the
+        foreseeable future.
+
+        If MalloryB's preimage spend was presigned, the only replacement
+        she could make would be to go from one feerate to a higher
+        feerate.  She couldn't add new inputs to the preimage spend, and
+        without that capability, she would be unable to initiate the
+        removal part of a replacement cycle.
+
+    - **OP_EXPIRE:** in a separate thread, but quoting from the
+      replacement cycles thread, Peter Todd [proposed][todd expire1]
+      several consensus changes to enable an `OP_EXPIRE` opcode will
+      make a transaction invalid for inclusion after a specified block
+      height if the transaction's script executes `OP_EXPIRE`.  This can
+      be used to make the Mallory's preimage condition of an HTLC only
+      usable before Bob's refund condition becomes spendable.  This
+      prevents Mallory from being able to replace Bob's refund spend,
+      making it impossible for Mallory to execute a replacement cycle
+      attack.  `OP_EXPIRE` may also address some [transaction pinning
+      attacks][topic transaction pinning] against HTLCs.
```

Comment:
Good answer.  I upvoted it.  I didn't mention that concern here, in this very short summary, because Peter Todd took steps to address that concern in his proposal (to the same degree that coinbase transactions address that problem).